src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -205,12 +221,51 @@ public DRSAccessURL getAccessUrlForObjectId(
                     new IllegalArgumentException(
                         "No matching access ID " + accessId + " was found on object " + objectId));
 
+    if (wrapper.isGcp()) {
+      return signGoogleUrl(snapshot, matchingAccessMethod);
+    } else {
+      return signAzureUrl(billingProfileModel, snapshot, drsId);
+    }
+  }
+
+  private Optional<DRSAccessMethod> getAccessMethodMatchingAccessId(
+      String accessId, DRSObject object) {
+    return object.getAccessMethods().stream()
+        .filter(
+            drsAccessMethod ->
+                drsAccessMethod.getType().equals(TypeEnum.GS)
+                    && drsAccessMethod.getAccessId().equals(accessId))
+        .findFirst();
+  }
+
+  private DRSAccessURL signAzureUrl(BillingProfileModel profileModel, Snapshot snapshot, DrsId drsId) {
+    FSItem fsObject = null;
+    try {
+      fsObject = fileService.lookupSnapshotFSItem(
+          snapshotService.retrieveAvailableSnapshotProject(snapshot.getId()), drsId.getFsObjectId(), 1);
+
+      BlobUrlParts blobUrl = BlobUrlParts.parse(fsObject.getPath());
+      BlobContainerClientFactory sourceClientFactory =
+          new BlobContainerClientFactory(
+              blobUrl.getAccountName(),
+              resourceConfiguration.getAppToken(UUID.fromString(profileModel.getTenantId())),
+              blobUrl.getBlobContainerName());
+
+      return new DRSAccessURL().url(sourceClientFactory.createReadOnlySasUrlForBlob(blobUrl.getBlobName()));
```

Comment:
Probably worth adding a parameter to the `createReadOnlySasUrlForBlob` method to be able to specify a timeout

Reply:
`BlobCrl` now has the ability to set a time out and create SAS tokens with the required permissions.